File: backend/app/models/whiteboard.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class Whiteboard(BaseModel):
    model_config = ConfigDict(populate_by_name=True)
    id: str = Field(alias="_id")
    workspace_id: str
    title: str
    board_data: dict
    created_by: str
    is_locked: bool = False
    locked_by: str | None = None
    locked_at: datetime | None = None
    created_at: datetime
    updated_at: datetime

File: frontend/src/types/whiteboard.ts
export type BoardObject = {
  id: string
  type: 'path' | 'rect' | 'circle' | 'arrow' | 'text'
  points?: number[]
  x?: number
  y?: number
  width?: number
  height?: number
  text?: string
  color: string
  rotation?: number
  fontSize?: number
}

export type Whiteboard = {
  id: string
  workspace_id: string
  title: string
  board_data: { objects: BoardObject[]; background: string; zoom: number }
  created_by: string
  is_locked: boolean
  locked_by: string | null
  locked_at: string | null
  created_at: string
  updated_at: string
}

export type EditorAccessRequest = {
  id: string
  workspace_id: string
  requester_id: string
  requester_name: string
  requester_email: string
  status: 'pending' | 'approved' | 'rejected'
  created_at: string
  handled_by: string | null
  handled_at: string | null
}

export type InviteToken = {
  id: string
  token: string
  workspace_id: string
  created_by: string
  expires_at: string
  revoked: boolean
  created_at: string
}
